src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.cs

```diff
@@ -43,5 +43,68 @@ public sealed override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationAction(CompilationAnalyzer.AnalyzeCompilation);
         }
+
+        protected bool IsNotSupportedByLiveAnalysis(int code)
+        {
+            if (code == MessageProvider.ERR_FailedToCreateTempFile ||
```

Comment:
I think it is going to be a *big* task for compiler team to maintain what diagnostics can be returned by compilation.GetDiagnostics and what can't.
I wouldn't say current approach of this analyzer is wrong, but rather conservative. Any analyzer can return hundreds of diagnostics that it never reports in any possible executable code path. It only guarantees I won't report diagnostics outside this set, we never have an exact set requirement in the API.